server/datastore/mysql/hosts.go

```diff
@@ -546,17 +546,31 @@ func (ds *Datastore) CountHosts(ctx context.Context, filter fleet.TeamFilter, op
 	return count, nil
 }
 
-func (ds *Datastore) CleanupIncomingHosts(ctx context.Context, now time.Time) error {
-	sqlStatement := `
+func (ds *Datastore) CleanupIncomingHosts(ctx context.Context, now time.Time) ([]uint, error) {
+	var ids []uint
+	selectIDs := `
+    SELECT
+      id
+    FROM
+      hosts
+    WHERE
+      hostname = '' AND
+      osquery_version = '' AND
+      created_at < (? - INTERVAL 5 MINUTE)`
+	if err := ds.writer.SelectContext(ctx, &ids, selectIDs, now); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "load incoming hosts to cleanup")
+	}
+
+	cleanupHosts := `
```

Comment:
This would potentially send a lot more data though (and I believe there is some limit to the number of values one can pass in a `IN (?)` condition). The statements are guaranteed to affect the same IDs, given that the time reference is static and the condition is based on the `created_at` timestamp (unless we'd start creating hosts in the past).

Reply:
gotcha, thanks for clarifying!